google/cloud/firestore_bundle/bundle.py

```diff
@@ -222,7 +222,8 @@ async def _process_async_query(
     ) -> datetime.datetime:
         doc: DocumentSnapshot
         _read_time = datetime.datetime.min.replace(tzinfo=UTC)
-        async for doc in snapshot.stream():
+        docs = await snapshot.stream()
+        async for doc in docs:
```

Comment:
we shouldn't have that extra await - the goal should be for this to be invisible to existing users

You should change `async def stream` to `def stream`, so it remains an AsyncGeneratorFunction, instead of becoming a Coroutine

Reply:
I don't think I explained this very well when we talked, so here's another try: 

- When `async` is used in the signature of a generator function, it tells Python to build an `AsyncGenerator` class instead of a standard `Generator`
- when `async` is used in standard function signatures, it says to wrap everything in a `Coroutine` that needs to be awaited
- We broke out the async generator into a helper `_stream` method. But we still want `_stream` to return an AsyncGenerator, not a coroutine. So we strip the `async` from the method signature, and turn it into a regular method. But it still behaves the same was as before

Note: this is also how the gapic generator generates [async streaming rpcs](https://github.com/googleapis/python-firestore/blob/9521deddc5a4b16956f37136f84928ac99688022/google/cloud/firestore_v1/services/firestore/async_client.py#L1182). They're standard functions that return async objects

Reply:
Thanks! Reverted these changes.